### PR TITLE
Updated Traefik to v2

### PR DESCRIPTION
**Pull request by applejag**
Traefik v1 is quite old and outdated. It got annoying to try and seek out the relevant old and deprecated documentation for Traefik v1, so I updated to Traefik v2.

There were backward incompatible changes in the configs, which I have updated according to this: https://doc.traefik.io/traefik/migration/v1-to-v2/

Some of the configs were also inspired by the examples over at https://github.com/containous/traefik-library-image#traefik-v2---example-usage

### CODE REVIEW

**Review by Alexamakans — APPROVED** [review]

File: docker-compose.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 version: '3'
 services:
   proxy:
-    image: traefik:1.7.10-alpine
+    image: amd64/traefik:v2.5.0
```
Comment (fredx30) [inline docker-compose.yml]: I have objections here. v2.5 is still an RC version. Im not objecting to using an RC version. We should just absolutely not be specifying a minor version. If the still expected bug to be comes along then the specification of the minor version ensures that we dont get the bug fix.

**Review by fredx30 — CHANGES_REQUESTED** [review]

**Review by iverestefans — APPROVED** [review]
LGTM!

File: docker-compose.yml (hunk 1 above)
Comment (applejag) [reply docker-compose.yml]: Don't know where you got the info that it's an RC? It's a regular release here: https://github.com/traefik/traefik/releases/tag/v2.5.0

And I completely forgot about the minor version tag thing, you're right! It's very suitable to skip the patch version in dev stuff like this. I'll change that

Comment (fredx30) [reply docker-compose.yml]: Yeah i stand corrected the release has been out for a full 2 days.

**Review by fredx30 — APPROVED** [review]

File: docker-compose.yml (hunk 1 above)
Comment (applejag) [reply docker-compose.yml]: Oh lol I didn't even look at the date. Fun! :p

**Pull request merged**